lib/container/kafka_container.ex

```diff
@@ -0,0 +1,264 @@
+    }
+  end
+
+  @doc """
+  Overrides the default image used for the Kafka container.
+  Right now we support only confluentinc images.
+  """
+  def with_image(%__MODULE__{} = config, image) when is_binary(image) do
+    %{config | image: image}
+  end
+
+  @doc """
+  Overrides the default kafka port used for the Kafka container.
+  """
+  def with_kafka_port(%__MODULE__{} = config, kafka_port) when is_integer(kafka_port) do
+    %{config | kafka_port: kafka_port}
+  end
+
+  @doc """
+  Overrides the default kafka port used for the Kafka container.
+  """
+  def with_broker_port(%__MODULE__{} = config, broker_port) when is_integer(broker_port) do
+    %{config | broker_port: broker_port}
+  end
+
+  @doc """
+  Overrides the default consensus strategy used for the Kafka container.
+  """
+  def with_consensus_strategy(%__MODULE__{} = config, consensus_strategy)
+      when consensus_strategy in [:zookeeper_embedded, :zookeeper_external, :kraft] do
+    %{config | consensus_strategy: consensus_strategy}
+  end
+
+  @doc """
+  Overrides the default zookeeper port used for the Kafka container.
+  """
+  def with_zookeeper_port(%__MODULE__{consensus_strategy: strategy} = config, zookeeper_port)
+      when is_integer(zookeeper_port) and strategy in [:zookeeper_embedded, :zookeeper_external] do
+    %{config | zookeeper_port: zookeeper_port}
+  end
+
+  @doc """
+  Overrides the default zookeeper host used for the Kafka container.
+  Available only when consensus_strategy is external
+  """
+  def with_zookeeper_host(
+        %__MODULE__{consensus_strategy: :zookeeper_external} = config,
+        zookeeper_host
+      )
+      when is_binary(zookeeper_host) do
+    %{config | zookeeper_host: zookeeper_host}
+  end
+
+  @doc """
+  Overrides the default zookeeper host used for the Kafka container.
+  Available only when consensus_strategy is kraft
+  """
+  def with_cluster_id(%__MODULE__{consensus_strategy: :kraft} = config, cluster_id)
+      when is_binary(cluster_id) do
+    %{config | cluster_id: cluster_id}
+  end
+
+  @doc """
+  Overrides the default wait timeout used for the Kafka container.
+  """
+  def with_wait_timeout(%__MODULE__{} = config, wait_timeout) when is_integer(wait_timeout) do
+    %{config | wait_timeout: wait_timeout}
+  end
+
+  @doc """
+  Overrides the default topic
+  """
+  def with_topic_partitions(%__MODULE__{} = config, topic_partitions)
+      when is_integer(topic_partitions) do
+    %{config | default_topic_partitions: topic_partitions}
+  end
+
+  defimpl Testcontainers.ContainerBuilder do
+    import Container
+
+    @impl true
+    @spec build(%KafkaContainer{}) :: %Container{}
+    def build(%KafkaContainer{} = config) do
+      new(config.image)
+      |> with_fixed_port(config.kafka_port)
+      |> with_environment(:KAFKA_BROKER_ID, "1")
+      |> with_listener_config(config)
+      |> with_topic_config(config)
+      |> with_startup_script(config)
```

Comment:
draft PR here: https://github.com/testcontainers/testcontainers-elixir/pull/50 
it fails because of issues with Tesla client and the fact I am not proficient with how to stream files with Tesla